### DISCUSSION TITLE
General Discussions

### DISCUSSION THREAD

**Discussion by Biswa96** [Biswa96/initrdg#1] [General]
Here it is.

@crramirez @Dakkaron @angelog0

**Comment by Dakkaron** [comment]
Nice! This is pretty awsome! Impressive, that it only took this "little" amount of code!

Pretty impressive work! I will try it out soon!

**Comment by crramirez** [comment]
Excellent thanks a lot for this work. I'll check it soon. Maybe it also help people on Microsoft to back port it

**Comment by angelog0** [comment]
Thanks. I hope MS back ports this to Win10.. their software states that my PC cannot upgrade to Win 11 becaus this processor, Ryzen 5 2400G is not supported...

**Comment by Biswa96** [comment]
@CarlosNihelton Are you working on the systemd + WSL idea? I have seen that in a [phoronix][1] post. FYI, this is a fun project to do something before the init process.

[1]: https://www.phoronix.com/scan.php?page=news_item&px=PoC-Ubuntu-WSL2-systemd

**Reply by cnihelton** [reply]
Very interesting work! Thanks for calling my attention. Right now my boss is working on the systemd. I'll share this repo with him.

**Reply by Biswa96** [reply]
Remember this repo is just for fun and testing purpose.

**Reply by cnihelton** [reply]
Sure!

**Comment by Biswa96** [comment]
Update! If I use desktop-shell instead of rdprail-shell then everything works. Tested with konsole, terminator etc. One major downside is that everything runs in a whole window instead of separate app window. Playing kapman with sound:

![Capture](https://user-images.githubusercontent.com/31443074/144111160-ee9781e0-61c2-4524-b686-e2f4dd7812ce.PNG)

The fonts are normal. I minimize the window for screenshot.

**Reply by Biswa96** [reply]
rdprail-shell may have some issue in Windows 10.

**Comment by hello-smile6** [comment]
Could WSL be installed without administrator privileges using this?

**Reply by Biswa96** [reply]
This repository is not about installing WSL. There are things in WSL core layer which need system level access like - installing lxcore kernel mode driver, registering lxss manager service, registering lxss user session COM server, copying things from WinSxS to System32 and the list goes on.

**Reply by hello-smile6** [reply]
Oh, could you make user-mode implementations of any drivers, use the path confusion bug to make Windows think the files in system32, and maybe avoid registering services by running the processes on login?

**Reply by Biswa96** [reply]
AFAIK, it is not possible. You can learn more about WSL internals here https://www.youtube.com/watch?v=_p3RtkwstNk

**Reply by hello-smile6** [reply]
😦

**Reply by Dakkaron** [reply]
It might be possible, but that's basically rewriting WSL.

**Reply by hello-smile6** [reply]
@Dakkaron Then why not? Does qemu have a library that works as a DLL? If not, someone could make one, or it could be used from JS. If qemu will work, people would just have to write a backend (maybe use node and qemu port forwarding) that'd be compatible with Microsoft's.

**Comment by Biswa96** [comment]
Does anyone here use the new Android subsystem on Windows? I am thinking to explore it with this project. I'm not sure if it is possible to run APK file but it is possibly play with system binaries. Just need output from some commands (e.g. mount) and event trace logs.

**Reply by hello-smile6** [reply]
Run Windows 11 in a virtual machine and start looking at memory dumps.

**Reply by Biswa96** [reply]
I can not enable virtualization features in VM. CPU too old.

**Reply by hello-smile6** [reply]
Somehow https://computernewb.com/collab-vm/ manages to run Windows 11 in a VM.

**Reply by Biswa96** [reply]
You misread my point. I can run Windows 11 in a VM but I can't enable virtualization in that VM for and WSL/WSA related stuff.

**Reply by hello-smile6** [reply]
Oh, could you enable virtualization in the guest and run a VM on that?

**Reply by Biswa96** [reply]
Nope.

**Reply by hello-smile6** [reply]
That's strange.
